engine/engine/src/engine.cpp

```diff
@@ -2017,11 +2017,59 @@ void dmEngineFinalize()
     dmSocket::Finalize();
 }
 
+bool GetGraphicsAdapterTypeFromArgs(int argc, char *argv[], dmGraphics::AdapterType* adapter_type_out)
+{
+    const char arg_adapter_type[] = "--graphics-adapter=";
+    for (int i = 0; i < argc; ++i)
+    {
+        const char* arg = argv[i];
+        if (strncmp(arg_adapter_type, arg, sizeof(arg_adapter_type)-1) == 0)
+        {
+            const char* eq = strchr(arg, '=');
+            if (strncmp("opengl", eq+1, sizeof("opengl")-1) == 0) {
+                *adapter_type_out = dmGraphics::ADAPTER_TYPE_OPENGL;
```

Comment:
I wonder if it could be the graphics api that returns a valid adapter type given a string?

```
const char* adapter_s = ParseArg("--graphics-adapter=", argc, argv);
if (adapter_s) {
    dmGraphics::AdapterType adapter_type;
    bool result = dmGraphics::GetAdapterType(adapter_s, &adapter_type);
}
```